Don't override user provided swagger definition options

brantw opened the pull request:
Only setting properties on the swaggerObject that do not exist to prevent overwriting user specified properties.

kalinchernev commented on lib/index.js:
```diff
@@ -145,6 +145,33 @@ function convertGlobPaths(globs) {
+ * @returns {object} Swagger Object with required properties set
+ */
+function populateMissingProperties(swaggerObject) {
+  if (!swaggerObject.hasOwnProperty('paths')) {
```
Just wondering, since these checks for creating empty objects in properties are repetitive. Wouldn't it be ok to make an array of required properties that are attached iteratively if not available?

efmr commented:
On my recent pull Request #30 I do also this but just for the tags.

amcdnl commented:
@kalinchernev Can this be merged now?

kalinchernev commented:
It's up to @drGrove or @chdanielmueller :)

drGrove commented:
@kalinchernev I'll merge this tonight

drGrove commented:
@kalinchernev, can you fix the merge issues with this against latest master?

kalinchernev commented:
Hi @drGrove I can take a look.
I have no write access, do I push to the branch or I shall follow another approach?

kalinchernev commented:
@drGrove I've opened a #34 pull request taking master of swimlane to keep credits of commits, but I am not confident enough to push it directly to the master.
Do you prefer to merge the develop first and then this one, or rather ask for an update in the swimlane master?

drGrove commented:
@kalinchernev, my mistake, @brantw can you fix this pull request against latest master?

brantw commented:
I rebased on top of latest and adjusted my changes to match @efmr's changes.

kalinchernev commented:
@brantw I thought you wanted to add populateMissingProperties() but you remove it in 5cd2bd4
Probably you chose the other way around of the suggested conflict resolution.
The end diff at the moment is only changing default values.

brantw commented:
@kalinchernev That was intentional.  The intent of the pull request was to prevent properties already set on `swaggerObject` from being overridden when default values are populated.  My first implementation was to check if the property existed before setting the default value and only setting a default value if the property did not already exist.   I probably didn't make this clear in my commit message, but when I rebased with master I changed my code to match what @efmr's did with the` tags` property to the rest of the properties on `swaggerObject` which I believe will achieve the same result and is much terser.
